src/OpenTelemetry/Instrumentation/ListenerHandler.cs

```diff
@@ -17,29 +17,63 @@
 
 namespace OpenTelemetry.Instrumentation
 {
+    /// <summary>
+    /// ListenerHandler base class.
+    /// </summary>
     public abstract class ListenerHandler
     {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ListenerHandler"/> class.
+        /// </summary>
+        /// <param name="sourceName">The name of the <see cref="ListenerHandler"/>.</param>
         public ListenerHandler(string sourceName)
         {
             this.SourceName = sourceName;
         }
 
+        /// <summary>
+        /// Gets the name of the <see cref="ListenerHandler"/>.
+        /// </summary>
         public string SourceName { get; }
 
+        /// <summary>
+        /// Gets a value indicating whether the <see cref="ListenerHandler"/> supports NULL <see cref="Activity"/>.
+        /// </summary>
         public virtual bool SupportsNullActivity { get; } = false;
 
+        /// <summary>
+        /// Method called when an <see cref="Activity"/> from this source is started.
```

Comment:
In thinking about this a bit more... `OnStart`, `OnStop`, etc. - `Method called when an <see cref="Activity"/> from this source is started.` is not quite accurate since it is dependent on the library author naming the events they produce with either a `Start` or `Stop` suffix.

Though, I'm finding it a little difficult to capture words that better capture this. Something like:
`Method called when this <see cref="ListenerHandler"/> responds to an event with the suffix 'Start'`

Or maybe your wording here is fine if others feel it is pretty common for library authors to conform to this convention. The convention is cited [here](https://github.com/dotnet/runtime/blob/master/src/libraries/System.Diagnostics.DiagnosticSource/src/DiagnosticSourceUsersGuide.md#event-names). But then that leaves the `OnException` and `OnCustom` callbacks that are not referenced there.

Reply:
Method called when this <see cref="ListenerHandler"/> responds to an event with the suffix 'Start'

This sounds more accurate.

Reply:
Thanks for the suggestion. For the OnException and OnCustom, do u have something in mind? Because they don't have a pattern just Start/Stop.